src/imgui/Shortcuts.cc

```diff
@@ -0,0 +1,136 @@
+});
+static_assert(allShortcutInfo.size() == ShortcutIndex::NUM_SHORTCUTS);
+// Note: if we forget a row we get a compile error.
+//       if we swap the order of some rows we also get a compile error (when asserts are enabled)
+
+static constexpr auto defaultShortcuts = []{
+    std::array<Shortcuts::Data, ShortcutIndex::NUM_SHORTCUTS> result = {};
+    for (int i = 0; i < ShortcutIndex::NUM_SHORTCUTS; ++i) {
+        auto& all = allShortcutInfo[i];
+        assert(all.index == i); // verify that table is in-order
+        result[i].keyChord = all.keyChord;
+        result[i].local    = all.local;
+        result[i].repeat   = all.repeat;
+    }
+    return result;
+}();
+
+static constexpr auto shortcutNames = []{
+    std::array<std::string_view, ShortcutIndex::NUM_SHORTCUTS> result = {};
+    for (int i = 0; i < ShortcutIndex::NUM_SHORTCUTS; ++i) {
+        result[i] = allShortcutInfo[i].name;
+    }
+    return result;
+}();
+
+static constexpr auto shortcutDescriptions = []{
+    std::array<std::string_view, ShortcutIndex::NUM_SHORTCUTS> result = {};
+    for (int i = 0; i < ShortcutIndex::NUM_SHORTCUTS; ++i) {
+        result[i] = allShortcutInfo[i].description;
+    }
+    return result;
+}();
+
+Shortcuts::Shortcuts(SettingsConfig& config)
+	: shortcuts{}
+{
+	config.setShortcuts(*this);
+	setDefaultShortcuts();
+}
+
+void Shortcuts::setDefaultShortcuts()
+{
+	int index = 0;
+	for (const auto& shortcut : defaultShortcuts) {
+		setShortcut(static_cast<ShortcutIndex>(index++), shortcut.keyChord, shortcut.local, shortcut.repeat);
+	}
+}
+
+Shortcuts::Data& Shortcuts::getShortcut(ShortcutIndex index)
+{
+	assert(index < ShortcutIndex::NUM_SHORTCUTS);
+	return shortcuts[index];
+}
+
+void Shortcuts::setShortcut(ShortcutIndex index, std::optional<ImGuiKeyChord> keyChord, std::optional<bool> local, std::optional<bool> repeat)
+{
+	assert(index < ShortcutIndex::NUM_SHORTCUTS);
+	if (keyChord) shortcuts[index].keyChord = *keyChord;
+	if (local) shortcuts[index].local = *local;
+	if (repeat) shortcuts[index].repeat = *repeat;
+}
+
+void Shortcuts::setDefaultShortcut(ShortcutIndex index)
+{
+	assert(index < ShortcutIndex::NUM_SHORTCUTS);
+	const auto& shortcut = defaultShortcuts[index];
+	setShortcut(index, shortcut.keyChord, shortcut.local, shortcut.repeat);
+}
+
+const std::string_view Shortcuts::getShortcutName(ShortcutIndex index)
+{
+	assert(index < ShortcutIndex::NUM_SHORTCUTS);
+	return shortcutNames[index];
+}
+
+std::optional<ShortcutIndex> Shortcuts::getShortcutIndex(const std::string_view name)
+{
+        auto it = ranges::find_if(allShortcutInfo, [&](const auto& p) { return p.name == name; });
+        if (it == allShortcutInfo.end()) return {};
+        return std::optional<ShortcutIndex>(static_cast<ShortcutIndex>(std::distance(allShortcutInfo.begin(), it)));
+}	
+
+const std::string_view Shortcuts::getShortcutDescription(ShortcutIndex index)
```

Comment:
Detail: returning a const-by-value type has no meaning. Just return a (non-const) `std::string_view`.
(Returning a const **reference** is useful, but this is not (and should not be) a reference.)